0000-macro-value.md

```diff
@@ -0,0 +1,300 @@
+- Feature Name: `macro_value`
+- Start Date: 2021-03-27
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Add a new macro_rules matcher, `$name:value`, with identical semantics to that of a function capture.
+
+# Motivation
+[motivation]: #motivation
+
+Value arguments to function-like macros are tricky to deal with.
+While macro_rules macros don't suffer from the common and most egregious pitfalls of C-style preprocessor macros,
+such as misnested brackets and operator precedence,
+using an `$:expr` capture more than once still evaluates the expression more than once,
+duplicating side effects.
+
+Additionally,
+we have the additional wrinkle of the lifetime and drop timing of temporaries complicating matters further,
+if your intent is to write a macro invocation with equivalent-to-function-call semantics.
+Suffice to say,
+`let arg = $arg;` has the incorrect drop behavior,
+and the current best practice is to instead expand to
+
+```rust
+match ( $arg0, $arg1, ) {
+    ( arg0, arg1, ) => { /* macro body */ }
+}
+```
+
+instead.
+We can simplify this and make getting the correct behavior easier on macro authors.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+(in a section explaining `macro_rules` matchers:)
```

Comment:
Spurious comment?

Reply:
This may or may not be useful, but it is intended, to indicate the context of _when_ a guide/learner would be expected to explain/learn about `$:value`.